### DISCUSSION TITLE
Object Detection，Segmentation fault: 11

### DISCUSSION THREAD

**Discussion by Enderfga** [autogluon/autogluon#1737] [Q&A] [answered]
this my code：
```
time_limit = 60*60*4  # at most 4 hour
detector = ObjectDetector()
detector.fit(dataset_train, time_limit=time_limit, presets='good_quality_fast_inference')
```
I have been running this code for 200min, and I have finished 50 epochs twice, and I got this error for the third time.
And what should I do to get better results for the target detection task? I found that using "best_quality" gives me an error

**Comment by Enderfga** [comment]
Also, I use the equipment for Ubuntu 18.04.

**Comment by bryanyzhu** [comment, marked as answer]
Hi, can you share more details, like the error message, dataset size (number of classes, total number of bboxes), etc? It seems you can train the model for 50 epochs, so the error is unlikely caused by the environment. 

To obtain better results, it also depends. If you have a good amount of data, then some HPO and a better detector should do the job. If you have few training data, then either consider label more data, or using a super large model to do few-shot fine-tuning.

**Reply by Enderfga** [reply]
Thank your for you replyl! It is really kind of you.
I'm sorry I can't provide any other error information, it was only this one line and ran for almost an hour without output.
I have 3200 images in my training set, of which there are 4 categories with variable number of bboxes.
I am a beginner in the direction of target detection, and there are many shortcomings, I found that all parameters except best_quality work fine, for example good_quality_faster_inference.
I ended up training a model with an mAP of about 0.6 using over 8000 seconds with that parameter.

**Reply by bryanyzhu** [reply]
mAP of 0.6 is quite poor considering your dataset only has 4 categories. `best_quality` will take longer, so it is not available. For now, you can try our old [tutorial](https://auto.gluon.ai/stable/tutorials/object_detection/beginner.html) to see if works for you. In terms of preparing dataset, you can follow this [tutorial](https://auto.gluon.ai/stable/tutorials/object_detection/dataset.html). Thanks.
